xcm/xcm-executor/src/assets.rs

```diff
@@ -140,22 +153,78 @@ impl Assets {
 	/// Return the assets in `self`, but (asset-wise) of no greater value than `assets`.
 	///
 	/// Result is undefined if `assets` includes elements which match to the same asset more than once.
-	pub fn min<'a, I: Iterator<Item=&'a MultiAsset>>(&self, assets: I) -> Self {
+	pub fn min<'a, M: 'a + std::borrow::Borrow<MultiAsset>, I: Iterator<Item=M>>(&self, assets: I) -> Self {
```

Comment:
I don't think it's a good idea to use `std` here. A more worrying issue is that the CI passes, therefore we don't have testing for non std environment